roles/docker/defaults/main.yml

```diff
@@ -3,10 +3,7 @@ docker_apt_repo: https://download.docker.com/linux
 debian_apt_repo: "{{ docker_apt_repo }}/debian"
 ubuntu_apt_repo: "{{ docker_apt_repo }}/ubuntu"
 
-docker_debian_ce_version: "18.03.1"
-docker_ubuntu_ce_version: "18.03.1"
-docker_centos_ce_version: "18.06.0"
 docker_ce_pkg_name: docker-ce
-debian_docker_ce_version: "{{ docker_debian_ce_version }}~ce-0~debian"
-ubuntu_docker_ce_version: "{{ docker_ubuntu_ce_version }}~ce~3-0~ubuntu"
-centos_docker_ce_version: "{{ docker_centos_ce_version }}.ce"
+debian_docker_ce_version: "{{ (ansible_distribution_release == 'buster') | ternary('5:20.10.7~3-0~debian-buster', '5:19.03.15~3-0~debian-stretch') }}"
```

Comment:
Just my two cents, for this sort of version variables, I'd rename them more specifically like `debian_buster_docker_ce_version` and `debian_stretch_ce_version` and then use the `when` clause to pick or skip the version in each task. Right now, the potential issue I can tell is you only check if the distribution is `buster` or not. If not, no matter the distribution is newer or older, this variable will be set to the stretch version silently. What do you think?